### PR TITLE
ETP-235

### PR DESCRIPTION
**Pull request by cliffordp**
[ETP-235]

* Consolidate duplicated code from child classes into parent class: `\Tribe__Tickets__Tickets::get_tickets()`
* Add a few little defensive bits of code to avoid calling provider classes that aren't callable

5min demo video: https://share.getcloudapp.com/04uKX4v0

[ETP-235]: https://moderntribe.atlassian.net/browse/ETP-235

### CODE REVIEW

File: src/Tribe/Tickets.php [hunk 1]
```diff
@@ -621,15 +621,48 @@ public function save_ticket( $post_id, $ticket, $raw_data = [] ) {
...
-		 * @abstract
+		 * @since TBD Changed from protected abstract to public with duplicated child classes' logic consolidated here.
 		 *
 		 * @param int $post_id ID of parent "event" post.
 		 *
 		 * @return Tribe__Tickets__Ticket_Object[] List of ticket objects.
 		 */
-		protected function get_tickets( $post_id ) {}
+		public function get_tickets( $post_id ) {
+			$default_provider = Tribe__Tickets__Tickets::get_event_ticket_provider( $post_id );
```
Comment (tr1b0t) [inline src/Tribe/Tickets.php:633]: :no_entry_sign: Must use "self::" for local static member reference

<sub>hash: 2e7ebe326ceb3a32cd407ea004627df4</sub>

**Comment by bordoni** [comment]
Can we get some description into this PR?

**Comment by cliffordp** [comment]
@bordoni, do you ever sleep? ;) It's there now (in both repos' PRs)

File: src/Tribe/Editor/Blocks/Tickets.php [hunk 2]
```diff
@@ -203,13 +203,20 @@ public function get_tickets( $post_id ) {
...
 		if ( ! $all_tickets ) {
-			return array();
+			return [];
 		}
 
-		$tickets = array();
+		/** @var Tribe__Tickets__RSVP $rsvp */
+		$rsvp = tribe( 'tickets.rsvp' );
 
+		$tickets = [];
```
Comment (sc0ttkclark) [inline src/Tribe/Editor/Blocks/Tickets.php:212]: Should this use `$rsvp->get_tickets( $post_id );` instead now?

Comment (cliffordp) [reply src/Tribe/Editor/Blocks/Tickets.php:212]: The description of `\Tribe__Tickets__Editor__Blocks__Tickets::get_tickets()` is:
> Get all tickets for event/post, removing RSVPs

And `\Tribe__Tickets__Editor__Blocks__Rsvp::get_tickets()` is _protected_.

Maybe we should use ORM to get for all active providers other than 'rsvp'???

Comment (sc0ttkclark) [reply src/Tribe/Editor/Blocks/Tickets.php:212]: Technically these should all be using the ORM for their specific provider. Can you make that change?

File: readme.txt [hunk 3]
```diff
@@ -123,6 +123,7 @@ Currently, the following add-ons are available for Event Tickets:
 * Fix - Update file path in the docblocks of the templates for The Events Calendar new views. [ETP-289]
 * Fix - Update position of the `tribe_tickets_ticket_add` action so it receives the ticket data on ticket creation. [ETP-302]
 * Fix - Correct the `Total` value, when the WooCommerce options for currency are empty (Thousand and decimal separator, number of decimals). [ETP-231]
+* Tweak - Consolidate duplicated code from child classes into parent class: `\Tribe__Tickets__Tickets::get_tickets()`. [ETP-235]
```
Comment (sc0ttkclark) [inline readme.txt:126]:
```suggestion
* Tweak - Consolidate duplicate code for getting tickets for each Ticket Provider. [ETP-235]
```

**Pull request merged**